### DISCUSSION TITLE
Can Viper work without a config file?

### DISCUSSION THREAD

**Discussion by rakshitgondwal** [spf13/viper#1764] [Q&A]
Hey! I had this one question that can Viper work without a config file? If yes, how does viper manage the configuration in that case? Where does it write or read the values from?

**Comment by onlinejudge95** [comment]
Did you find any solution to this?
I am looking for a use case where the env vars are not loaded from `.env` file, typical case in container deployments, whereas `viper` reading the env vars and storing it in config `struct` like `pydantic`

**Reply by rakshitgondwal** [reply]
Not yet.

**Reply by onlinejudge95** [reply]
I found one solution where you can let go of reading from any config file and instead read from env vars, but the current implementation of `viper` would force you to bind all your envs 1st, I was expecting viper to register all the vars from `os.Environ()` implicitly.

Maybe try something like
```golang
package main

import (
    "fmt"
    "log"

    "github.com/spf13/viper"
)

type (
    Config struct {
        Name string `mapstructure:"NAME"`
        Host string `mapstructure:"HOST"`
    }
)

func main() {
    viper.SetEnvPrefix("APP")
    viper.AutomaticEnv()
    viper.BindEnv("NAME")
    viper.BindEnv("HOST")
    config := Config{}
    fmt.Println(viper.AllSettings())
    if err := viper.Unmarshal(&config); err != nil {
        log.Fatalln(err)
    }
    fmt.Printf("%#v\n", config)
}
```
